src/PowerShellEditorServices/Workspace/WorkspaceFileSystemWrapper.cs

```diff
@@ -0,0 +1,328 @@
+                    e);
+                yield break;
+            }
+            catch (PathTooLongException e)
+            {
+                this.logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to the path being too long",
+                    e);
+                yield break;
+            }
+            catch (Exception e) when (e is SecurityException || e is UnauthorizedAccessException)
+            {
+                this.logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to the path not being accessible",
+                    e);
+                yield break;
+            }
+            catch (Exception e)
+            {
+                this.logger.WriteHandledException(
+                    $"Could not enumerate files in the path '{dirInfo.FullName}' due to an exception",
+                    e);
+                yield break;
+            }
+            if (result == null) { yield break; }
+            foreach (var item in result)
+            {
+                if (this.ignoreReparsePoints && item.Attributes.HasFlag(FileAttributes.ReparsePoint)) { continue; }
+                // If it's a directory then automatically return it
+                if (item.Attributes.HasFlag(FileAttributes.Directory)) { yield return item; continue; }
+                if (this.allowedExtensions == null || this.allowedExtensions.Length == 0) { yield return item; continue; }
+                foreach (string extension in allowedExtensions)
+                {
+                    if (item.Extension == extension) { yield return item; break; }
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Wraps an instance of <see cref="System.IO.DirectoryInfo" /> and provides implementation of
+    /// <see cref="Microsoft.Extensions.FileSystemGlobbing.Abstractions.DirectoryInfoBase" />.
+    /// Based on https://github.com/aspnet/Extensions/blob/c087cadf1dfdbd2b8785ef764e5ef58a1a7e5ed0/src/FileSystemGlobbing/src/Abstractions/DirectoryInfoWrapper.cs
+    /// </summary>
+    public class WorkspaceFileSystemDirectoryWrapper : DirectoryInfoBase
+    {
+        private readonly DirectoryInfo concreteDirectoryInfo;
+        private readonly bool isParentPath;
+        private readonly WorkspaceFileSystemWrapperFactory factory;
+        private readonly int depth;
+
+        /// <summary>
+        /// Initializes an instance of <see cref="WorkspaceFileSystemDirectoryWrapper" />.
+        /// </summary>
+        public WorkspaceFileSystemDirectoryWrapper(WorkspaceFileSystemWrapperFactory factory, DirectoryInfo directoryInfo, int depth)
+        {
+            this.concreteDirectoryInfo = directoryInfo;
+            this.isParentPath = (depth == 0);
+            this.factory = factory;
+            this.depth = depth;
+        }
+
+        /// <inheritdoc />
```

Comment:
what does this do? Use the virtual's docs?

Reply:
Uses the parent class' documentation